modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,399 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Panel from 'Panel';
+import Loader from 'Loader';
+import swal from 'sweetalert2';
+
+/**
+ * Create Timepoint.
+ *
+ * @description form for create timepoint.
+ *
+ * @author Alizée Wickenheiser
+ * @version 1.0.0
+ *
+ */
+class CreateTimepoint extends React.Component {
+  /**
+   * @constructor
+   * @param {object} props - React Component properties
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {
+        pscid: '',
+        dccid: '',
+        visit: '',
+        languages: '',
+      },
+      form: {
+        display: {
+          subproject: false,
+          visit: false,
+          psc: false,
+          project: false,
+          languages: false,
+        },
+        options: {
+          subproject: {},
+          visit: {},
+          psc: {},
+          project: {},
+          languages: {},
+          required: {
+            subproject: true,
+            psc: true,
+            project: true,
+            visit: true,
+            languages: true,
+          },
+        },
+        value: {
+          subproject: null,
+          visit: null,
+          psc: null,
+          project: null,
+          languages: null,
+        },
+      },
+      storage: {
+        visit: {},
+      },
+      messages: [],
+      url: {
+        params: {
+          candID: '',
+          identifier: '',
+        },
+      },
+      success: false,
+    };
+
+    // Bind component instance to custom methods
+    this.fetchInitializerData = this.fetchInitializerData.bind(this);
+    this.handleVisitLabel = this.handleVisitLabel.bind(this);
+    this.collectParams = this.collectParams.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setForm = this.setForm.bind(this);
+  }
+
+  /**
+   * Executes after component mounts.
+   */
+  componentDidMount() {
+    this.collectParams().then(() => {
+      this.fetchInitializerData().then(() => {
+        this.setState({isLoaded: true});
+      });
+    });
+  }
+
+  /**
+   * Retrieve params from the browser URL and save it in state.
+   */
+  async collectParams() {
+    const url = new URL(window.location.href);
+    const state = JSON.parse(JSON.stringify(this.state));
+    state.url.params = {
+      candID: url.searchParams.get('candID'),
+      identifier: url.searchParams.get('identifier'),
+    };
+    state.data.dccid = state.url.params.candID;
+    await new Promise((resolve) => this.setState(state, resolve));
```

Comment:
I think this should work... although resolve need to be called as a function, i.e. `resolve()`

Reply:
I'm not sure if this is really making full use of `async/await` though, and I think it could be written like this with the same effect:
```
  collectParams() {
    const url = new URL(window.location.href);
    const state = JSON.parse(JSON.stringify(this.state));
    state.url.params = {
      candID: url.searchParams.get('candID'),
      identifier: url.searchParams.get('identifier'),
    };
    state.data.dccid = state.url.params.candID;
    return new Promise((resolve) => this.setState(state, resolve()));
```

Reply:
@HenriRabalais I'm a little confused as I don't think the `resolve()` is necessary instead of `resolve`. 

I think calling `resolve` with no parenthesis is just calling the method but not the value of that method and calling `resolve()` with parenthesis is calling the value of that method.

A few of the answers found here: https://stackoverflow.com/questions/43370176/using-async-setstate don't use the parentheses either.

I can change it to `resolve()` if you want but waiting on what you think about the above.

Reply:
Sorry, my mistake, resolve doesn't need to be called here! this.setState will take care of calling the function upon state update.